securedrop_client/gui/widgets.py

```diff
@@ -1150,26 +1150,32 @@ def update(self):
                 self.paperclip.hide()
             self.star.update()
         except sqlalchemy.exc.InvalidRequestError as e:
-            logger.error(
-                "Could not update SourceWidget for source %s: %s",
-                self.source.uuid,
-                e
-            )
+            logger.error(f"Could not update SourceWidget for source {self.source_uuid}: {e}")
             raise
 
-    def set_snippet(self, source, uuid=None, content=None):
+    def set_snippet(self, source, uuid=None, content=None) -> None:
         """
         Update the preview snippet only if the new message is for the
         referenced source and there's a source collection. If a uuid and
         content are passed then use these, otherwise default to whatever the
         latest item in the conversation might be.
         """
-        if source == self.source.uuid and self.source.collection:
-            last_collection_object = self.source.collection[-1]
-            if uuid and uuid == last_collection_object.uuid and content:
-                self.preview.setText(content)
-            else:
-                self.preview.setText(str(last_collection_object))
+        if source != self.source_uuid:
+            return
+
+        self.preview.setText("")
+
+        try:
+            self.controller.session.refresh(self.source)
+            if self.source.collection:
```

Comment:
if you want to reduce one layer of nesting you could just return if self.source.collection is empty up top instead